site/content/integrate/apps/functionality/external-webhooks/_index.md

```diff
@@ -7,87 +7,98 @@ aliases:
-| `APP_ID`         | The App's ID as found in the [manifest]({{<ref "/integrate/apps/structure/manifest">}})  | `my-app`                        |
-| `SUB_PATH`       | An arbitrary endpoint path defined by the App                                            | `my-endpoint`                   |
-| `WEBHOOK_SECRET` | A secret key generated by the Mattermost server that is used to authenticate the request | `9a44ckeqytd3bftn3c3y53968o`    |
+| Parameter name        | Description                                                                             | Example value                   |
+|-----------------------|-----------------------------------------------------------------------------------------|---------------------------------|
+| `mattermost_site_url` | The base URL of the Mattermost server.                                                  | `https://my-mattermost-server/` |
+| `app_id`              | The App's ID as found in the [manifest]({{<ref "/integrate/apps/structure/manifest">}}) | `my-app`                        |
+| `sub_path`            | An arbitrary path defined by the App                                                    | `my-webhook-path`               |
 
 Using the above example values, the webhook URL will look like this:
 
+`http://my-mattermost-server/plugins/com.mattermost.apps/apps/my-app/webhook/my-webhook-path`
 
-`http://my-mattermost-server/plugins/com.mattermost.apps/apps/my-apps/webhook/my-endpoint?secret=9a44ckeqytd3bftn3c3y53968o`
+The `/webhook` call is made every time an App webhook is accessed. The `sub_path` of the webhook URL can be obtained from the `path` property of the call request.
 
-### Retrieve the secret key
+## Webhook authentication
 
-When the App is first installed, the `OnInstall` [call]({{<ref "/integrate/apps/structure/call">}}) handler will receive a context value of `app.webhook_secret` in the request. This value should be persisted by the App for authenticating incoming webhook requests.
+App webhooks support an API key-like authentication method. A Mattermost server-generated `webhook_secret` is provided during App installation and is used by the implementer to validate a webhook request.
+
+There are two steps to enable this authentication method:
+
+1. Set the `remote_webhook_auth_type` property of the App manifest to `secret`.
+2. Define an `on_install` [call]({{<ref "/integrate/apps/structure/call">}}) in the App manifest with the `app` expand property set to `all`.
+   The `webhook_secret` property of the `app` context field in the call request contains the webhook secret.
 
 {{<note>}}
-The `OnInstall` call must have the `app` expand field set to `summary` for the generated secret key to be populated in the call request.
+The webhook secret should be stored by the App for authenticating incoming webhook requests. The App [key-value store]({{<ref "/integrate/apps/functionality/kv-store">}}) is suitable for storing the webhook secret. 
```

Comment:
The framework automatically authenticates the webhook requests, so Apps should not be storing the secret for this purpose. That's covered in this other PR as well though so we can reconcile there if you'd like https://github.com/mattermost/mattermost-developer-documentation/pull/1184

Reply:
I will remove this note.

Reply:
Do you think we should note that the framework automatically checks the secret on webhook requests though?

Reply:
I've noted that as part of App webhook authentication.